### PR TITLE
Fixes and additions to NL80211 interface

### PR DESCRIPTION
**Pull request by XDjackieXD**
This PR adds parsing for the NL80211_RATE_INFO_ and NL80211_STA_BSS_PARAM_ enums and fixes parsing of the NL80211_STA_FLAG_ array.

### CODE REVIEW

**Comment by svinota** [comment]
Thanks a lot, testing!

File: pyroute2/netlink/nl80211/__init__.py [hunk 1]
```diff
@@ -1216,44 +1216,97 @@ def decode(self):
...
-                    )
+                offset = self.offset + 4
+                mask, set_ = struct.unpack_from(
+                    'II', self.data, offset
+                )
 
-                    if mask & NL80211_STA_FLAG_AUTHORIZED:
-                        if set_ & NL80211_STA_FLAG_AUTHORIZED:
-                            self.value["AUTHORIZED"] = True
+                if mask & (1 << NL80211_STA_FLAG_AUTHORIZED):
```
Comment (svinota) [inline pyroute2/netlink/nl80211/__init__.py:1224]: Could it be better to move this shift to the flag definition like it is done for `NL80211_SCAN_FLAG_LOW_PRIORITY` etc?

**Review by svinota — CHANGES_REQUESTED** [review]

**Comment by svinota** [comment]
Please run also `make format` so `black` will fix the codestyle, but it's not a blocker, I can do it myself.

**Comment by XDjackieXD** [comment]
Done

**Comment by svinota** [comment]
LGTM, merging shortly

**Review by svinota — APPROVED** [review]

**Comment by svinota** [comment]
Thanks for contribution!

**Pull request merged**